src/Azure.Deployments.Extensibility.Core/V2/Models/Resource.cs

```diff
@@ -45,5 +46,7 @@ public Resource(string type, string? apiVersion, TIdentifiers identifiers, TProp
         public required TProperties Properties { get; init; }
 
         public string? Status { get; init; } = null;
+        
+        public required string ConfigId { get; init; }
```

Comment:
Should this be nullable instead? Not all extensions require the config object. For those extensions it may make sense for them to return null.

Reply:
Fixed.

I updated this further to add the checksum validation on CreateOrUpdate/Get/Preview but only if it's provided in the request payload. It can be utilized generally to ensure a target control plane.